Cpp_hands_on_programming/src/ipc_sendfile.cc

```diff
@@ -0,0 +1,21 @@
+/*
+ * ipc_sendfile.c
+ *
+ *  Created on: Feb 17, 2022
+ *      Author: kt
+*/
+#include "lib/other/Prog_init.h"
+
+int main(int argc, char* argv[]) {
+	try {
+		std::shared_ptr<Ipc_info> info;
+		Prog_init start;
+		start.printInstruction();
+		info = start.checkOptions(argc, argv);
+		start.run_IPC(info, SEND);
+	}
+	catch(const std::exception &e) {
+		std::cout << "Error: " << e.what() << std::endl;
```

Comment:
You may consider re-throwing exception (std::rethrow_exception), so you application won't return 0 in case of failure

Reply:
Yea, I don't think the process should return 0 after failure. You can return -1 as you said, but re-thrown will be more similar to just not catching exception (which would honestly be preferred way to me, but catch needs to be here to ensure stack unwinding)